xclim/cli.py

```diff
@@ -0,0 +1,304 @@
+@click.pass_context
+def info(ctx, indicator):
+    """Give information about INDICATOR."""
+    for indname in indicator:
+        ind = _get_indicator(indname)
+        command = _create_command(indname)
+        formatter = click.HelpFormatter()
+        with formatter.section(
+            click.style("Indicator", fg="blue")
+            + click.style(f" {indname}", fg="yellow")
+        ):
+            data = ind.json()
+            data.pop("parameters")
+            _format_dict(data, formatter, key_fg="blue", spaces=2)
+
+        command.format_options(ctx, formatter)
+
+        click.echo(formatter.getvalue())
+
+
+def _format_dict(data, formatter, key_fg="blue", spaces=2):
+    for attr, val in data.items():
+        if isinstance(val, list):
+            for isub, sub in enumerate(val):
+                formatter.write_text(
+                    click.style(" " * spaces + f"{attr} (#{isub})", fg=key_fg)
+                )
+                _format_dict(sub, formatter, key_fg=key_fg, spaces=spaces + 2)
+        elif isinstance(val, dict):
+            formatter.write_text(click.style(" " * spaces + f"{attr}:", fg=key_fg))
+            _format_dict(val, formatter, key_fg=key_fg, spaces=spaces + 2)
+        else:
+            formatter.write_text(
+                click.style(" " * spaces + attr + " :", fg=key_fg) + " " + str(val)
+            )
+
+
+class XclimCli(click.MultiCommand):
+    """Main cli class."""
+
+    def list_commands(self, ctx):
+        """Return the available commands (other than the indicators)."""
+        return "indices", "info"
+
+    def get_command(self, ctx, name):
+        """Return the requested command."""
+        command = {"indices": indices, "info": info}.get(name)
+        if command is None:
+            command = _create_command(name)
+        return command
+
+
+@click.command(
+    cls=XclimCli,
+    chain=True,
+    help="Command line tool to compute indices on netCDF datasets. Indicators are reffered to by their (case-insensitive) identifier, as in xclim.core.indicator.registry.",
+    invoke_without_command=True,
+    subcommand_metavar="INDICATOR1 [OPTIONS] ... [INDICATOR2 [OPTIONS] ... ] ...",
+)
+@click.option(
+    "-i",
+    "--input",
+    help="Input files. Can be a netCDF path or a glob pattern.",
+    multiple=True,
+)
+@click.option("-o", "--output", help="Output filepath. A new file will be created")
+@click.option("-v", "--verbose", help="Make it more verbose", count=True)
+@click.option("--version", is_flag=True, help="Prints xclim's version number and exits")
+@click.option(
+    "--dask-nthreads",
+    type=int,
+    help="Start a dask.distributed Client with this many threads and 1 worker. If not specified, the local schedular is used. If specified, '--dask-maxmem' must also be given",
+)
+@click.option(
+    "--dask-maxmem",
+    help="Memory limit for the dask.distributed Client as a human readable string (ex: 4GB). If specified, '--dask-nthreads' must also be specified.",
```

Comment:
It felt like this would be extremely machine-dependent and that I wouldn't be able to find a good generic default number?  Like 4GB would be safe for small machines but completely useless on large ones...

Reply:
Would establishing a baseline from a percentage make sense? For example, report available memory, take half and round to the nearest GB?

I imagine this might be overkill and would probably break if it encounters wacky OSes or server infrastructure. hmmm...

Reply:
Ok, then let's leave it at that and see how it goes. Maybe keep @Zeitsperre suggestion for later.